packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -1010,45 +900,150 @@ private boolean updateMapStyle(String style) {
     // Dart passes an empty string to indicate that the style should be cleared.
     final MapStyleOptions mapStyleOptions =
         style == null || style.isEmpty() ? null : new MapStyleOptions(style);
-    final boolean set = Objects.requireNonNull(googleMap).setMapStyle(mapStyleOptions);
-    lastStyleError =
-        set ? null : "Unable to set the map style. Please check console logs for errors.";
-    return set;
+    lastSetStyleSucceeded = Objects.requireNonNull(googleMap).setMapStyle(mapStyleOptions);
+    return lastSetStyleSucceeded;
+  }
+
+  /** MapsApi implementation */
+  @Override
+  public void waitForMap(@NonNull Messages.VoidResult result) {
+    if (googleMap == null) {
+      mapReadyResult = result;
+    } else {
+      result.success();
+    }
+  }
+
+  @Override
+  public @NonNull Messages.PlatformPoint getScreenCoordinate(
+      @NonNull Messages.PlatformLatLng latLng) {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized",
+          "getScreenCoordinate called prior to map initialization",
+          null);
+    }
+    Point screenLocation =
+        googleMap.getProjection().toScreenLocation(Convert.latLngFromPigeon(latLng));
+    return Convert.pointToPigeon(screenLocation);
+  }
+
+  @Override
+  public @NonNull Messages.PlatformLatLng getLatLng(
+      @NonNull Messages.PlatformPoint screenCoordinate) {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getLatLng called prior to map initialization", null);
+    }
+    LatLng latLng =
+        googleMap.getProjection().fromScreenLocation(Convert.pointFromPigeon(screenCoordinate));
+    return Convert.latLngToPigeon(latLng);
+  }
+
+  @Override
+  public @NonNull Messages.PlatformLatLngBounds getVisibleRegion() {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getVisibleRegion called prior to map initialization", null);
+    }
+    LatLngBounds latLngBounds = googleMap.getProjection().getVisibleRegion().latLngBounds;
+    return Convert.latLngBoundsToPigeon(latLngBounds);
+  }
+
+  @Override
+  public @NonNull Double getZoomLevel() {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getZoomLevel called prior to map initialization", null);
```

Comment:
I added this, since unlike most of the other methods there was no handling, so there was a possible-null-reference warning in Android Studio. I've been cleaning up warnings as I move code.